### ISSUE TITLE
I can not download any course

### ISSUE THREAD

**Issue by omarcumx** [liranbg/linkedin-learning-downloader#20]
![image](https://user-images.githubusercontent.com/16604468/58366456-315fc700-7e98-11e9-953c-1b5184568a1a.png)
C:\link\linkedin>python linkedin_learning.py
2019-05-25 02:52:00,888 asyncio      DEBUG    Using selector: SelectSelector
2019-05-25 02:52:00,891 root         INFO     [*] -------------Login------------
-
2019-05-25 02:52:00,892 root         INFO     [*] Login step 1 - Getting CSRF to
ken...
Traceback (most recent call last):
  File "linkedin_learning.py", line 226, in <module>
    loop.run_until_complete(process())
  File "C:\Users\Server\AppData\Local\Programs\Python\Python36\lib\asyncio\base_
events.py", line 466, in run_until_complete
    return future.result()
  File "linkedin_learning.py", line 210, in process
    await login(USERNAME, PASSWORD)
  File "linkedin_learning.py", line 82, in login
    csrf = html.xpath("//input[@id='loginCsrfParam-login']/@value").pop()
IndexError: pop from empty list

> this is the error, do you know how to solve it?

**Comment by jochemke** [comment]
I have the same error. Are there any details about this?

**Comment by emobile-dev** [comment]
I am also facing the similar issue on windows 10 with python 3.7.2

**Comment by jaaufauvre** [comment]
Replace:
```python
csrf = html.xpath("//input[@id='loginCsrfParam-login']/@value").pop()
```
with: 
```python
csrf = html.xpath("//input[@name='loginCsrfParam']/@value").pop()
```

**Comment by liranbg** [comment]
resolved #21

**Comment by jcrawford** [comment]
I am getting this error as well, does not seem to be resolved.  It seems that they have a rate limit in place as I am getting back an http 429 error.  I am guessing I just downloaded too much too fast and they have blocked me.  Hopefully they remove it soon and it's not permanent.
